### PR TITLE
tooltip formlabel

### PR DESCRIPTION
**Pull request by jpp-odoo**

### CODE REVIEW

File: addons/web/static/src/legacy/xml/base.xml [hunk 1]
```diff
@@ -439,7 +439,7 @@
         <t t-esc="info.field.help"/>
     </p>
 
-    <ul class="o-tooltip--technical" role="tooltip">
+    <ul class="o-tooltip--technical" t-if="info.debug" role="tooltip">
```
Comment (aab-odoo) [inline addons/web/static/src/legacy/xml/base.xml]: legacy??

File: addons/web/static/src/views/form/form_label.js [hunk 2]
```diff
@@ -16,9 +16,42 @@ export class FormLabel extends Component {
...
+        const activeField = this.props.record.activeFields[this.props.fieldName];
+
+        const info = {
+            viewMode: "form",
+            resModel: this.props.record.resModel,
+            debug: Boolean(odoo.debug),
+            field: {
+                label: this.props.string,
+                name: this.props.fieldName,
+                noLabel: activeField.noLabel,
```
Comment (aab-odoo) [inline addons/web/static/src/views/form/form_label.js]: noLabel, in the tooltip of a label?

File: addons/web/static/src/views/form/form_label.js [hunk 3]
```diff
@@ -16,9 +17,49 @@ export class FormLabel extends Component {
...
+                help: this.tooltipHelp,
+                type: field.type,
+                widget: activeField.widget,
+                // widgetXMLName: column.FieldComponent.XMLName, //WOWL: Todo
```
Comment (jpp-odoo) [inline addons/web/static/src/views/form/form_label.js]: - remove from list and form (and fieldChart)

File: addons/web/static/src/views/form/form_label.js [hunk 4]
```diff
@@ -16,9 +17,49 @@ export class FormLabel extends Component {
...
+                // widgetDescription: column.FieldComponent.description || linkedField.description, //WOWL: Todo
```
Comment (jpp-odoo) [inline addons/web/static/src/views/form/form_label.js]: activeField.component.displayName

File: addons/web/static/src/views/form/form_label.js [hunk 5]
```diff
@@ -16,9 +17,49 @@ export class FormLabel extends Component {
...
+                context: activeField.context,
+                domain: activeField.domain.toString(),
+                modifiers: JSON.stringify(activeField.modifiers),
+                //change_default: linkedField.change_default, //WOWL: Todo
```
Comment (jpp-odoo) [inline addons/web/static/src/views/form/form_label.js]: field

File: addons/web/static/src/views/form/form_label.js [hunk 6]
```diff
@@ -16,9 +17,49 @@ export class FormLabel extends Component {
...
+                // relation: linkedField.relation, //WOWL: Todo
+                // selection: linkedField.selection, //WOWL: Todo
```
Comment (jpp-odoo) [inline addons/web/static/src/views/form/form_label.js]: field

File: addons/web/static/src/views/form/form_label.js [hunk 7]
```diff
@@ -16,9 +17,49 @@ export class FormLabel extends Component {
...
+            },
```
Comment (jpp-odoo) [inline addons/web/static/src/views/form/form_label.js]: default ? in field ?

File: addons/web/static/src/views/form/form_label.js [hunk 8]
```diff
@@ -16,9 +17,49 @@ export class FormLabel extends Component {
...
     }
+
+    get hasBigTooltip() {
+        return Boolean(odoo.debug) || this.tooltipHelp;
+    }
+
+    get tooltipHelp() {
+        const field = this.props.record.fields[this.props.fieldName];
+        let help = field.help || "";
+        if (field.company_dependent && session.display_switch_company_menu) {
```
Comment (jpp-odoo) [inline addons/web/static/src/views/form/form_label.js:28]: Pass display_... And do the Values set here are ... in the template

File: addons/web/static/src/views/form/form_label.js [hunk 9]
```diff
@@ -16,9 +17,49 @@ export class FormLabel extends Component {
         }
         return classes.join(" ");
     }
+
+    get hasBigTooltip() {
+        return Boolean(odoo.debug) || this.tooltipHelp;
```
Comment (jpp-odoo) [inline addons/web/static/src/views/form/form_label.js:22]:
```suggestion
        return Boolean(odoo.debug) || (field.company_dependent && session.display_switch_company_menu);
```

File: addons/web/static/src/fields/field_tooltip.js [hunk 10]
```diff
@@ -0,0 +1,37 @@
+/** @odoo-module **/
+
+import fieldRegistry from "web.field_registry";
+
+export function tooltipInfo(params) {
+    let widgetDescription = undefined;
+    if (params.activeField.widget) {
+        if (params.activeField.FieldComponent.name === "LegacyField") {
+            const widget = fieldRegistry.get(params.activeField.widget);
+            widgetDescription = widget.prototype.description;
```
Comment (SimonGenin) [inline addons/web/static/src/fields/field_tooltip.js:10]: is `widget.prototype.description` the same as `widget.constructor.description` ?

Comment (aab-odoo) [reply addons/web/static/src/fields/field_tooltip.js:10]: it's the way to retrieve "static" properties on legacy classes

File: addons/web/static/src/fields/field_tooltip.js [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+            label: params.field.string,
+            name: params.field.name,
+            help: params.help || params.field.help,
+            type: params.field.type,
+            widget: params.activeField.widget,
+            widgetDescription,
+            context: params.activeField.context,
+            domain: params.activeField.domain.toString(),
+            modifiers: JSON.stringify(params.activeField.modifiers),
+            change_default: params.field.change_default,
```
Comment (aab-odoo) [inline addons/web/static/src/fields/field_tooltip.js]: camelcase

File: addons/web/static/src/fields/field_tooltip.js [hunk 12]
```diff
@@ -0,0 +1,37 @@
+/** @odoo-module **/
+
+import fieldRegistry from "web.field_registry";
+
+export function tooltipInfo(params) {
```
Comment (aab-odoo) [inline addons/web/static/src/fields/field_tooltip.js]: maybe rename into getTooltipInfo ?

File: addons/web/static/src/fields/field_tooltip.xml [hunk 13]
```diff
@@ -0,0 +1,73 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<templates xml:space="preserve">
+
+    <t t-name="web.FieldTooltip" owl="1">
```
Comment (aab-odoo) [inline addons/web/static/src/fields/field_tooltip.xml:4]: is this template new in master-wowl? I'm a bit concerned that it could be used by legacy code

**Review by aab-odoo — APPROVED** [review]
good job

File: addons/web/static/src/fields/field_tooltip.xml (hunk 13 above)
Comment (SimonGenin) [reply addons/web/static/src/fields/field_tooltip.xml:4]: it's new, the older one has another name

Comment (SimonGenin) [reply addons/web/static/src/fields/field_tooltip.xml:4]: the older one is `WidgetLabel.tooltip`

**Pull request merged**